pkgs/development/compilers/pakcs/default.nix

```diff
@@ -1,25 +1,39 @@
-{ lib, stdenv, fetchurl, makeWrapper
+{ lib, stdenv, fetchurl, fetchpatch2, makeWrapper
 , haskellPackages, haskell
 , which, swi-prolog, rlwrap, tk
 , curl, git, unzip, gnutar, coreutils, sqlite }:
 
 let
   pname = "pakcs";
-  version = "3.6.0";
+  version = "3.7.2";
 
   # Don't switch to "Current release" without a reason, because its
   # source updates without version bump. Prefer last from "Older releases" instead.
   src = fetchurl {
     url = "https://www.informatik.uni-kiel.de/~pakcs/download/pakcs-${version}-src.tar.gz";
-    hash = "sha256-1r6jEY3eEGESKcAepiziVbxpIvQLtCS6l0trBU3SGGo=";
+    hash = "sha256-ZfQUgFqmPPCeDx/T5G/JdvYDq/7XbvsgxPcEX4y9HZ4=";
   };
 
   curry-frontend = (haskellPackages.override {
     overrides = self: super: {
-      curry-frontend = haskell.lib.compose.overrideCabal (drv: {
-        inherit src;
-        postUnpack = "sourceRoot+=/frontend";
-      }) (super.callPackage ./curry-frontend.nix { });
+      curry-frontend = lib.pipe (super.callPackage ./curry-frontend.nix { })
+        [ haskell.lib.doJailbreak
+          (haskell.lib.compose.overrideCabal (drv: {
+            inherit src;
+            postUnpack = "sourceRoot+=/frontend";
+
+            # NOTE: We cannot use `appendPatches` because it conflicts with `overrideCabal`
```

Comment:
`appendPatches` should do the same as you have written out here. Any specific problems?

Reply:
Oh ok `haskell.lib.compose.appendPatch` works, I've tried `haskell.lib.appendPatch` before and it doesn't compose well with `overrideCabal` I guess
```
error:
       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:34:12:
           33|
           34|   strict = derivationStrict drvAttrs;
             |            ^
           35|

       … while evaluating derivation 'pakcs-3.7.2'
         whose name attribute is located at /home/t4ccer/repos/github/NixOS/nixpkgs/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'makeFlags' of derivation 'pakcs-3.7.2'
         at /home/t4ccer/repos/github/NixOS/nixpkgs/pkgs/development/compilers/pakcs/default.nix:43:3:
           42|
           43|   makeFlags = [
             |   ^
           44|     "CURRYFRONTEND=${curry-frontend}/bin/curry-frontend"

       (stack trace truncated; use '--show-trace' to show the full, detailed trace)

       error: attribute 'override' missing
       at /home/t4ccer/repos/github/NixOS/nixpkgs/pkgs/development/haskell-modules/lib/compose.nix:38:28:
           37|    */
           38|   overrideCabal = f: drv: (drv.override (args: args // {
             |                            ^
           39|     mkDerivation = drv: (args.mkDerivation drv).override f;
           ```

Reply:
I've pushed this change, doesn't even trigger a rebuild